src/Microsoft.Extensions.Configuration.Abstractions/IConfiguration.cs

```diff
@@ -36,5 +36,6 @@ public interface IConfiguration
         IEnumerable<IConfigurationSection> GetChildren();
 
         IChangeToken GetReloadToken();
+        IConfigurationRoot Root { get; }
```

Comment:
With this change IConfigurationRoot will also have a Root property, so with the default application template Intellisense on `Configuration` will show an almost meaningless property to millions of customers :smile:

Can we add the `Root` property only on the `IConfigurationSection` interface instead? We can always define the `GetRoot(IConfiguration)` extension method to do the right thing and put it in a namespace that isn't in the default template so that you opt in to see it?